uaclient/security.py

```diff
@@ -0,0 +1,397 @@
+            title=cve.title,
+            url_path="{}.json".format(cve.id.lower()),
+        )
+    )
+    affected_pkg_versions = {}
+    for source_pkg, package_status in cve.packages_status.items():
+        if source_pkg in installed_packages:
+            if package_status.status != "not-affected":
+                affected_pkg_versions[source_pkg] = package_status
+    return affected_pkg_versions
+
+
+def prompt_for_affected_packages(
+    client, issue_id, affected_pkg_status, installed_packages
+) -> None:
+    """Process security CVE dict returning a CVEStatus object.
+
+    Since CVEs point to a USN if active, get_notice may be called to fill in
+    CVE title details.
+    """
+    count = len(affected_pkg_status)
+    if len(affected_pkg_status) == 0:
+        print(
+            status.MESSAGE_SECURITY_AFFECTED_PKGS.format(
+                count="No", plural_str="s are"
+            )
+        )
+        print(status.MESSAGE_SECURITY_ISSUE_UNAFFECTED.format(issue=issue_id))
+        return
+
+    if len(affected_pkg_status) == 1:
+        plural_str = " is"
+    else:
+        plural_str = "s are"
+    msg = status.MESSAGE_SECURITY_AFFECTED_PKGS.format(
+        count=count, plural_str=plural_str
+    )
+    msg_suffix = ": " + ", ".join(affected_pkg_status.keys())
+    msgs = [msg + msg_suffix]
+    for pkg_num, pkg_name in enumerate(affected_pkg_status.keys(), 1):
+        msgs.append("({}/{}) {}:".format(pkg_num, count, pkg_name))
+        pkg_status = affected_pkg_status[pkg_name]
+        if pkg_status.status == "needed":
+            msgs.append("Sorry, no fix available yet")
+        elif pkg_status.status == "pending":
+            msgs.append("A fix is coming soon. Try again tomorrow.")
+        elif pkg_status.status in ("ignored", "deferred"):
+            msgs.append("Sorry, no fix is available.")
+        elif pkg_status.status == "released":
+            if pkg_status.pocket:
+                if pkg_status.pocket == "esm-infra":
+                    fix_stream = "UA Infra"
+                elif pkg_status.pocket == "esm-apps":
+                    fix_stream = "UA Apps"
+                elif pkg_status.pocket in ("updates", "security"):
+                    fix_stream = "Ubuntu standard updates"
+                else:
+                    fix_stream = "Unknown"
+            else:
+                fixed_version = pkg_status.description
+                if "esm" in fixed_version:
+                    fix_stream = "UA Infra"
+                else:
+                    fix_stream = "Ubuntu standard updates"
+            msgs.append(
+                status.MESSAGE_SECURITY_FIX_RELEASE_STREAM.format(
+                    fix_stream=fix_stream
+                )
+            )
+            try:  # get_pkg_install_status_messages
+                util.subp(
+                    [
+                        "dpkg",
+                        "--compare-versions",
+                        fixed_version,
```

Comment:
Maybe I am missing something here, but is seems that the `fixed_version` package will only be set if we do not have a pocket on the `pkg_status.status` object